### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by Ugboaja-Uchechi**
###  Project Requirements

- Create a class Corrector with a method correct_name that does the following:

  Make sure the first letter of the given name is a capital letter.

  Make sure that the name has a maximum of 10 characters. If it's longer it should trim the word.

- Set an instance of Corrector in Person on initialization.

- In Person add a method validate_name that uses the Corrector instance to validate and save the corrected name.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require './corrector'
+
```
Comment (ZeenatLawal) [inline person.rb:2]: - [x] In the Person class initialize method, an instance of Corrector should be set as stated in the requirements. Kindly add an instance of the Corrector class, which should be something like this `@corrector = Corrector.new()`

File: person.rb [hunk 2]
```diff
@@ -17,5 +19,9 @@ def can_use_services?
     of_age? || @parent_permission == true
   end
 
+  def validate_name
+    @corrector.correct_name(name)
```
Comment (ZeenatLawal) [inline person.rb]: - [x] I suggest you assign the corrected name to the `@name` instance to ensure the corrected name is being saved. It should look similar to this `@name = @corrector.correct_name(@name)`

**Review by ZeenatLawal — CHANGES_REQUESTED** [review]
Hi @Ugboaja-Uchechi, 

Good job so far!:clap: :clap:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎊 🎊 
- Git flow is followed. ✔️
- PR has a title and summary. ✔️


## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**